submarine-sdk/pysubmarine/submarine/tracking/client.py

```diff
@@ -101,24 +109,53 @@ def save_model(
         :param artifact_path: Relative path of the artifact in the minio pod.
         :param registered_model_name: If not None, register model into the model registry with
                                       this name. If None, the model only be saved in minio pod.
+        :param input_dim: Save the input dimension of the given model to the description file.
+        :param output_dim: Save the output dimension of the given model to the description file.
         """
         pattern = r"[0-9A-Za-z][0-9A-Za-z-_]*[0-9A-Za-z]|[0-9A-Za-z]"
         if not re.fullmatch(pattern, artifact_path):
             raise Exception(
-                "Artifact_path must only contains numbers, characters, hyphen and underscore.      "
-                "        Artifact_path must starts and ends with numbers or characters."
+                "Artifact_path must only contains numbers, characters, hyphen and underscore. "
+                "Artifact_path must starts and ends with numbers or characters."
             )
         with tempfile.TemporaryDirectory() as tempdir:
+            description: Dict[str, Any] = dict()
+            model_save_dir = os.path.join(tempdir, "1")
+            os.mkdir(model_save_dir)
             if model_type == "pytorch":
                 import submarine.models.pytorch
 
-                submarine.models.pytorch.save_model(model, tempdir)
+                submarine.models.pytorch.save_model(model, model_save_dir)
+                description["model_type"] = "pytorch"
```

Comment:
We are using triton to serve the PyTorch model therefore we need to use the `torch.jit.trace` to save the model. We must make it required for saving the PyTorch model since the `torch.jit.trace` function requires the input shape of the model.